### PR TITLE
[noTicket][RISK=NO]Send user demographic survey attribute Gender identity to RDR

### PR DESCRIPTION
**Pull request by NehaBroad**
**PR checklist**

- [ ] This PR meets the Acceptance Criteria in the JIRA story
- [ ] The JIRA story has been moved to Dev Review
- [ ] This PR includes appropriate unit tests
- [X ] I have run and tested this change locally
- [ X] If this includes an API change, I have updated the appropriate Swagger definitions and notified API consumers
- [ ] If this includes a new feature flag, I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later

### CODE REVIEW

File: api/src/main/resources/rdr.yaml [hunk 1]
```diff
@@ -234,12 +234,11 @@ definitions:
...
     enum:
-      - MALE
-      - FEMALE
+      - MAN
+      - WOMAN
       - NON_BINARY
       - TRANSGENDER
-      - INTERSEX
-      - NONE
+      - NONE_DESCRIBE_ME
```
Comment (jaycarlton) [inline api/src/main/resources/rdr.yaml]: Yeah, I like this better. It'll be cleaner definitely.

File: api/src/main/java/org/pmiops/workbench/rdr/RdrExportServiceImpl.java [hunk 2]
```diff
@@ -200,6 +200,12 @@ private RdrResearcher toRdrResearcher(DbUser dbUser) {
                 .map(RdrExportEnums::sexAtBirthToRdrSexAtBirth)
                 .collect(Collectors.toList()));
       }
+      if (null != dbDemographicSurvey.getGenderIdentityEnumList()) {
+        researcher.setGender(
+            dbDemographicSurvey.getGenderIdentityEnumList().stream()
+                .map(RdrExportEnums::genderToRdrGender)
+                .collect(Collectors.toList()));
+      }
```
Comment (jaycarlton) [inline api/src/main/java/org/pmiops/workbench/rdr/RdrExportServiceImpl.java]: nit: It takes a bit of getting used to, but I think it's cleaner to do
```
Optional.ofNullable(dbDemographicSurvey.getGenderIdentityEnumList())
    .map(g -> g.stream().map(RdrExportEnums::genderToRdrGender).collect(Collectors.toList()))
    .ifPresent(researcher::setGender);
```

`if`, `null`, and `!=` should generally be minimized IMO.

Comment (NehaBroad) [reply api/src/main/java/org/pmiops/workbench/rdr/RdrExportServiceImpl.java]: @jaycarlton : Do you mind taking another look?

File: api/src/main/resources/rdr.yaml [hunk 3]
```diff
@@ -281,10 +279,10 @@ definitions:
...
       - MASTER
       - DOCTORATE
-      -
+
   Disability:
     type: string
     enum:
-      - YES
-      - NO
+      - "YES"
```
Comment (NehaBroad) [inline api/src/main/resources/rdr.yaml]: There is a bug in Swagger that converts ENUM YES and NO to true and false unless its Explicitly used as string "YES"

File: api/src/main/java/org/pmiops/workbench/rdr/RdrExportEnums.java [hunk 4]
```diff
@@ -70,8 +72,8 @@
   private static final BiMap<Disability, org.pmiops.workbench.rdr.model.Disability>
       CLIENT_TO_RDR_DISABILITY =
           ImmutableBiMap.<Disability, org.pmiops.workbench.rdr.model.Disability>builder()
-              .put(Disability.TRUE, org.pmiops.workbench.rdr.model.Disability.TRUE)
-              .put(Disability.FALSE, org.pmiops.workbench.rdr.model.Disability.FALSE)
+              .put(Disability.TRUE, org.pmiops.workbench.rdr.model.Disability.YES)
```
Comment (jaycarlton) [inline api/src/main/java/org/pmiops/workbench/rdr/RdrExportEnums.java]: This confuses me. Either we have a boolean, in which case we don't need an enum, or we have an enum that describes  the situation like `Disability.NOT_DISABLED`, `Disability.SOME`, etc. But if you want to start an enum, it's hard to imagine what you'd add to yes and no.

Comment (NehaBroad) [reply api/src/main/java/org/pmiops/workbench/rdr/RdrExportEnums.java]: We basically have three options for DISABILITY FOR RDR its YES, NO ,PREFER_NOT_TO_ANSWER

File: api/src/main/java/org/pmiops/workbench/rdr/RdrExportServiceImpl.java [hunk 5]
```diff
@@ -194,26 +195,31 @@ private RdrResearcher toRdrResearcher(DbUser dbUser) {
...
-          dbDemographicSurvey.getRaceEnum().stream()
-              .map(RdrExportEnums::raceToRdrRace)
-              .collect(Collectors.toList()));
-    } else {
-      researcher.setRace(new ArrayList<>());
+          Optional.ofNullable(
+                  dbDemographicSurvey.getRaceEnum().stream()
+                      .map(RdrExportEnums::raceToRdrRace)
+                      .collect(Collectors.toList()))
+              .orElse(new ArrayList<Race>()));
```
Comment (jaycarlton) [inline api/src/main/java/org/pmiops/workbench/rdr/RdrExportServiceImpl.java]: Do you want the race field to be written to, or should the set of entries be fixed?

Clients might do `researcher.getRace().add(Race.DAYTONA_500);`, which works, but is counterintuitive.

File: api/src/main/resources/rdr.yaml (hunk 3 above)
Comment (jaycarlton) [reply api/src/main/resources/rdr.yaml]: 😠 

Can you please double-check that when this goes through our script to create `merged.yaml` that the quotes are still present? I'm worried they won't be.

In fact, I'm confused Swagger can distinguish the difference; by the time the YAML is loaded all bets are off w/r/t quotes.

File: api/src/main/java/org/pmiops/workbench/rdr/RdrExportServiceImpl.java [hunk 6]
```diff
@@ -194,26 +195,31 @@ private RdrResearcher toRdrResearcher(DbUser dbUser) {
...
-            dbDemographicSurvey.getSexAtBirthEnum().stream()
-                .map(RdrExportEnums::sexAtBirthToRdrSexAtBirth)
-                .collect(Collectors.toList()));
-      }
+      researcher.setSexAtBirth(
+          Optional.ofNullable(
+                  dbDemographicSurvey.getSexAtBirthEnum().stream()
+                      .map(RdrExportEnums::sexAtBirthToRdrSexAtBirth)
+                      .collect(Collectors.toList()))
+              .orElse(new ArrayList<SexAtBirth>()));
```
Comment (jaycarlton) [inline api/src/main/java/org/pmiops/workbench/rdr/RdrExportServiceImpl.java]: Looks good. Passing in `.ifPresent(researcher::setSexAtBirth)` is more a matter of taste.

File: api/src/main/java/org/pmiops/workbench/rdr/RdrExportServiceImpl.java (hunk 5 above)
Comment (NehaBroad) [reply api/src/main/java/org/pmiops/workbench/rdr/RdrExportServiceImpl.java]: no RDR needs to simple take whatever value we give to them and not manipulate them

File: api/src/main/resources/rdr.yaml (hunk 3 above)
Comment (NehaBroad) [reply api/src/main/resources/rdr.yaml]: Since this is in RDR.yaml, merged.yaml doesnt have these changes

**Pull request merged**